tests/api/reports-export.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/**
+ * Reports Export REST API Test
+ *
+ * @package WooCommerce\Tests\API
+ * @since 3.5.0
+ */
+
+/**
+ * Class WC_Tests_API_Reports_Export
+ */
+class WC_Tests_API_Reports_Export extends WC_REST_Unit_Test_Case {
+	/**
+	 * Export route.
+	 *
+	 * @var string
+	 */
+	protected $export_route = '/wc/v4/reports/(?P<type>[a-z]+)/export';
+
+	/**
+	 * Export status route.
+	 *
+	 * @var string
+	 */
+	protected $status_route = '/wc/v4/reports/(?P<type>[a-z]+)/export/(?P<export_id>[a-z0-9]+)/status';
+
+	/**
+	 * Setup test reports categories data.
+	 *
+	 * @since 3.5.0
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		$this->user = $this->factory->user->create(
+			array(
+				'role' => 'administrator',
+			)
+		);
+	}
+
+	/**
+	 * Test route registration.
+	 *
+	 * @since 3.5.0
+	 */
+	public function test_register_routes() {
+		$routes = $this->server->get_routes();
+
+		$this->assertArrayHasKey( $this->export_route, $routes );
+		$this->assertArrayHasKey( $this->status_route, $routes );
+	}
+
+	/**
+	 * Test the export of a taxes report.
+	 *
+	 * @since 3.5.0
+	 */
+	public function test_taxes_report_export() {
+		global $wpdb;
+		wp_set_current_user( $this->user );
+		WC_Helper_Reports::reset_stats_dbs();
+
+		// Populate all of the data.
+		$product = new WC_Product_Simple();
+		$product->set_name( 'Test Product' );
+		$product->set_regular_price( 25 );
+		$product->save();
+
+		// Add a GA tax rate.
+		$ga_rate_id = WC_Tax::_insert_tax_rate(
+			array(
+				'tax_rate'          => '7',
+				'tax_rate_country'  => 'US',
+				'tax_rate_state'    => 'GA',
+				'tax_rate_name'     => 'GA Tax',
+				'tax_rate_priority' => 1,
+				'tax_rate_order'    => 1,
+			)
+		);
+
+		// Add a FL tax rate.
+		$fl_rate_id = WC_Tax::_insert_tax_rate(
+			array(
+				'tax_rate'          => '6',
+				'tax_rate_country'  => 'US',
+				'tax_rate_state'    => 'FL',
+				'tax_rate_name'     => 'FL Tax',
+				'tax_rate_priority' => 1,
+				'tax_rate_order'    => 1,
+			)
+		);
+
+		// Create a GA order.
+		$order = WC_Helper_Order::create_order( 1, $product );
+		$order->set_billing_city( 'Savannah' );
+		$order->set_billing_state( 'GA' );
+		$order->set_billing_postcode( '31401' );
+		$order->set_status( 'completed' );
+		$order->calculate_totals();
+		$order->save();
+
```

Comment:
Would this line be referred to as the _Florida Georgia Line_ in the code? :trollface: 🇺🇸

Reply:
![](https://media3.giphy.com/media/13GKP7xGjce5oI/giphy.gif)